content/how-to/long-term-caching.md

```diff
@@ -0,0 +1,258 @@
+
+```html
+<html>
+  <head>
+    <script>
+    //<![CDATA[
+    window.webpackManifest = {"0":"main.3d038f325b02fdee5724.js","1":"1.c4116058de00860e5aa8.js"}
+    //]]>
+    </script>
+  </head>
+  <body>
+  </body>
+</html>
+```
+
+So the final webpack.config.js should look like this:
+
+```js
+var path = require('path');
+var webpack = require('webpack');
+var ManifestPlugin = require('webpack-manifest-plugin');
+var ChunkManifestPlugin = require('chunk-manifest-webpack-plugin');
+var WebpackMd5Hash = require('webpack-md5-hash');
+
+module.exports = {
+  entry: {
+    vendor: './src/vendor.js',
+    main: './src/index.js'
+  },
+  output: {
+    path: path.join(__dirname, 'build'),
+    filename: '[name].[chunkhash].js',
+    chunkFilename: '[name].[chunkhash].js'
+  },
+  plugins: [
+    new webpack.optimize.CommonsChunkPlugin({
+      name: "vendor",
+      minChunks: Infinity,
+    }),
+    new WebpackMd5Hash(),
+    new ManifestPlugin(),
+    new ChunkManifestPlugin({
+      filename: "chunk-manifest.json",
+      manifestVariable: "webpackManifest"
+    }),
+    new webpack.optimize.OccurenceOrderPlugin()
+  ]
+};
+
+```
+
+Or, if you're using [webpack-html-plugin](https://github.com/ampedandwired/html-webpack-plugin), you can use [inline-manifest-webpack-plugin](https://github.com/szrenwei/inline-manifest-webpack-plugin) to do this.
+
+Using this config the vendor chunk should not be changing its hash unless you change its code or dependencies. Here is a sample output for 2 runs with `moduleB.js` being changed between the runs:
+
+```
+> webpack
+
+Hash: 92670583f688a262fdad
+Version: webpack 1.10.1
+Time: 65ms
+
+Asset Size Chunks Chunk Names
+chunk-manifest.json 68 bytes [emitted]
+vendor.6d107863983028982ef4.js 3.71 kB 0 [emitted] vendor
+1.c4116058de00860e5aa8.js 107 bytes 1 [emitted]
+main.5e17f4dff47bc1a007c0.js 373 bytes 2 [emitted] main
+
+[0] ./src/index.js 186 bytes {2} [built]
+[0] ./src/vendor.js 40 bytes {0} [built]
+[1] ./src/moduleA.js 28 bytes {2} [built]
+[2] ./src/moduleB.js 28 bytes {1} [built]
+
+> webpack
+
+Hash: a9ee1d1e46a538469d7f
+Version: webpack 1.10.1
+Time: 67ms
+
+Asset Size Chunks Chunk Names
+chunk-manifest.json 68 bytes [emitted]
+vendor.6d107863983028982ef4.js 3.71 kB 0 [emitted] vendor
+1.2883246944b1147092b1.js 107 bytes 1 [emitted]
+main.5e17f4dff47bc1a007c0.js 373 bytes 2 [emitted] main
+
+[0] ./src/index.js 186 bytes {2} [built]
+[0] ./src/vendor.js 40 bytes {0} [built]
+[1] ./src/moduleA.js 28 bytes {2} [built]
+[2] ./src/moduleB.js 28 bytes {1} [built]
+```
+
```

Comment:
That looks cool. I've used [LICEcap](http://www.cockos.com/licecap/) myself. Ideally we would get gif output as that avoids a dependency on a third party service.